src/client/components/Dashboard/my-companies/MyCompaniesTable.jsx

```diff
@@ -0,0 +1,137 @@
+import { orderBy } from 'lodash'
+import moment from 'moment'
+import React from 'react'
+import styled from 'styled-components'
+import LinesEllipsis from 'react-lines-ellipsis'
+import { typography } from '@govuk-react/lib'
+import Table from '@govuk-react/table'
+import Link from '@govuk-react/link'
+import { GREY_1, GREY_3, TEXT_COLOUR } from 'govuk-colours'
+import { MEDIA_QUERIES } from '@govuk-react/constants'
+import Button from '@govuk-react/button'
+import useMyCompaniesContext from './useMyCompaniesContext'
+import Filters from './MyCompaniesFilters'
+
+const StyledCellHeader = styled(Table.CellHeader)(
+  typography.font({ size: 14, weight: 'bold' }),
+  {
+    [MEDIA_QUERIES.DESKTOP]: {
+      overflow: 'hidden',
+      'white-space': 'nowrap',
+    },
+  }
+)
+
+const StyledButtonTableCell = styled(Table.CellHeader)`
+  white-space: nowrap;
+`
+
+const StyledDateCell = styled(Table.Cell)(typography.font({ size: 14 }), {
+  color: GREY_1,
+})
+
+const StyledLink = styled.a`
+  margin-bottom: 0;
+`
+
+function sortCompanies(companies, sortType) {
+  const sort = {
+    recent: orderBy(
+      companies,
+      [(c) => c.latestInteraction.date || ''],
+      ['desc']
+    ),
+    'least-recent': orderBy(
+      companies,
+      [(c) => c.latestInteraction.date || ''],
+      ['asc']
+    ),
+    alphabetical: orderBy(companies, [(c) => c.company.name], ['asc']),
+  }
+  return sort[sortType]
```

Comment:
I suggest to use `if` or `switch` statement here to avoid unnecessary processing, e.g.

```suggestion
  switch (sortType) {
    case 'recent':
      return orderBy(
        companies,
        [(c) => c.latestInteraction.date || ''],
        ['desc']
      ),
    case 'least-recent':
      return orderBy(
        companies,
        [(c) => c.latestInteraction.date || ''],
        ['asc']
      ),
    case 'alphabetical':
      return orderBy(companies, [(c) => c.company.name], ['asc']),
```

We should also avoid magic strings, therefore `sortType` should be represented as a constant.